cw-dex-astroport/tests/astroport_tests.rs

```diff
@@ -141,48 +196,68 @@ mod tests {
+    fn test_withdraw_liquidity(
+        pool_type: PairType,
+        initial_liquidity: Vec<(&str, u64)>,
+        use_liquidity_manager: bool,
+    ) {
         let owned_runner = get_test_runner();
         let runner = owned_runner.as_ref();
-        let (accs, lp_token_addr, _pair_addr, contract_addr, asset_list, _) =
-            setup_pool_and_testing_contract(&runner, pool_type, initial_liquidity).unwrap();
+        let (accs, lp_token, pair_addr, contract_addr, asset_list, _) =
+            setup_pool_and_testing_contract(
+                &runner,
+                pool_type.clone(),
+                use_liquidity_manager,
+                initial_liquidity,
+            )
+            .unwrap();
         let admin = &accs[0];
         let wasm = Wasm::new(&runner);
 
-        //Query admin LP token balance
-        let admin_lp_token_balance =
-            cw20_balance_query(&runner, lp_token_addr.clone(), admin.address()).unwrap();
-        let amount_to_send = admin_lp_token_balance / Uint128::from(2u128);
+        let lp_token = if lp_token.starts_with(admin.prefix()) || lp_token.starts_with("contract") {
+            AssetInfo::cw20(Addr::unchecked(&lp_token))
+        } else {
+            AssetInfo::native(lp_token.clone())
+        };
+
+        let admin_lp_token_balance = query_asset_balance(&runner, &lp_token, &admin.address());
 
+        let amount_to_send = admin_lp_token_balance / Uint128::from(2u128);
         // Send LP tokens to contract
-        cw20_transfer(
+        send_asset(
             &runner,
-            lp_token_addr.clone(),
+            Asset::new(lp_token.clone(), amount_to_send),
             contract_addr.clone(),
-            amount_to_send,
             admin,
-        )
-        .unwrap();
-        let contract_lp_token_balance =
-            cw20_balance_query(&runner, lp_token_addr.clone(), contract_addr.clone()).unwrap();
+        );
+
+        let contract_lp_token_balance = query_asset_balance(&runner, &lp_token, &contract_addr);
         assert_eq!(contract_lp_token_balance, amount_to_send);
 
+        let withdraw_amount = contract_lp_token_balance / Uint128::from(2u128);
+
         // Simulate withdraw liquidity to get expected out assets
         let simulate_query = QueryMsg::SimulateWithdrawLiquidty {
-            amount: contract_lp_token_balance,
+            amount: withdraw_amount,
         };
         let expected_out: AssetList = wasm.query(&contract_addr, &simulate_query).unwrap();
 
         // Withdraw liquidity with min_out one more than expected_out. Should fail.
-        let unwrap = Unwrap::Err("but expected");
+        let error_msg = match &pool_type {
+            PairType::Custom(t) if t == "concentrated" => {
+                "Min out is not supported for concentrated liquidity pools"
+            }
+            _ => "but expected",
```

Comment:
why only part of expected message?

Reply:
idk ask the person who wrote that (probably me 2 years ago)